Orange/widgets/unsupervised/owmanifoldlearning.py

```diff
@@ -246,8 +247,11 @@ def apply(self):
                 projector = method(**self.get_method_parameters())
                 try:
                     self.progressBarSet(20)
-                    X = projector(self.data).embedding_
-                    data = Table(domain, X, self.data.Y, self.data.metas)
+                    if self.data.is_sparse():
```

Comment:
Could we move this check into `set_data` and not even call apply?
Currently if I click around in the widget progress bar starts, gets set to 20% and then we show the error.